src/Security/Authenticator/AdminTokenAuthenticator.php

```diff
@@ -0,0 +1,113 @@
+<?php
+declare(strict_types=1);
+
+/**
+ * This source file is available under the terms of the
+ * Pimcore Open Core License (POCL)
+ * Full copyright and license information is available in
+ * LICENSE.md which is distributed with this source code.
+ *
+ *  @copyright  Copyright (c) Pimcore GmbH (https://www.pimcore.com)
+ *  @license    Pimcore Open Core License (POCL)
+ */
+
+namespace Pimcore\Bundle\StudioBackendBundle\Security\Authenticator;
+
+use JsonException;
+use Pimcore\Bundle\StudioBackendBundle\Authorization\Controller\TokenLoginController;
+use Pimcore\Bundle\StudioBackendBundle\Exception\Api\AccessDeniedException;
+use Pimcore\Bundle\StudioBackendBundle\Exception\Api\InvalidArgumentException;
+use Pimcore\Security\User\User;
+use Pimcore\Tool\Authentication;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+use Symfony\Component\Security\Core\Exception\AuthenticationException;
+use Symfony\Component\Security\Core\User\UserInterface;
+use Symfony\Component\Security\Http\Authenticator\AbstractAuthenticator;
+use Symfony\Component\Security\Http\Authenticator\Passport\Badge\UserBadge;
+use Symfony\Component\Security\Http\Authenticator\Passport\Passport;
+use Symfony\Component\Security\Http\Authenticator\Passport\SelfValidatingPassport;
+use function sprintf;
+
+/**
+ * @internal
+ */
+class AdminTokenAuthenticator extends AbstractAuthenticator
+{
+    /**
+     *  @throws InvalidArgumentException
+     */
+    public function supports(Request $request): ?bool
+    {
+        return ($request->attributes->get('_route') === TokenLoginController::ROUTE_NAME) &&
+            $this->getTokenFromRequest($request) !== null;
+    }
+
+    /**
+     *  @throws AccessDeniedException|InvalidArgumentException
+     */
+    public function authenticate(Request $request): Passport
+    {
+        $pimcoreUser = Authentication::authenticateToken(
+            $this->getTokenFromRequest($request),
+        );
+
+        if ($pimcoreUser === null) {
+            throw new AccessDeniedException('Failed to authenticate with username and token');
```

Comment:
```suggestion
            throw new AccessDeniedException('Failed to authenticate with the given token');
```

We only provide the token and not a username?